[ERROR] Could not resolve "@mui/x-internals/fastMemo"

### Steps to reproduce

Steps:
1. Open this link to live example: https://stackblitz.com/edit/vitejs-vite-fc9449sy
2. Wait until dependencies installed and Vite development server started and wait a little longer
3. The following error message should be displayed _in the terminal_:

```
✘ [ERROR] Could not resolve "@mui/x-internals/fastMemo"

    node_modules/.pnpm/@mui+x-license@7.24.0_@types+react@18.3.18_react@18.3.1/node_modules/@mui/x-license/Watermark/Watermark.js:2:25:
      2 │ import { fastMemo } from '@mui/x-internals/fastMemo';
        ╵                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@mui/x-internals/fastMemo" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.

Error: Build failed with 1 error:
node_modules/.pnpm/@mui+x-license@7.24.0_@types+react@18.3.18_react@18.3.1/node_modules/@mui/x-license/Watermark/Watermark.js:2:25: ERROR: Could not resolve "@mui/x-internals/fastMemo"
    at failureErrorWithLog (/home/projects/vitejs-vite-fc9449sy/node_modules/.pnpm/esbuild@0.24.2/node_modules/esbuild/lib/main.js:1466:15)
    at buildResponseToResult/< (/home/projects/vitejs-vite-fc9449sy/node_modules/.pnpm/esbuild@0.24.2/node_modules/esbuild/lib/main.js:935:44)
    at runOnEndCallbacks/< (/home/projects/vitejs-vite-fc9449sy/node_modules/.pnpm/esbuild@0.24.2/node_modules/esbuild/lib/main.js:1344:13) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}

Node.js v18.20.3
 ELIFECYCLE  Command failed with exit code 1.
```

### Current behavior

Using `pnpm`, when I set `hoist=false` in my `.npmrc`, `vite` fails with the error described in the reproduction steps.

This suggests that `@mui/x-externals` is not listed as a dependency in `@mui/x-license` but it should be.

### Expected behavior

No error.

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
  Binaries:
    Node: 18.20.5 - /run/user/1000/fnm_multishells/58046_1737374688393/bin/node
    npm: 10.8.2 - /run/user/1000/fnm_multishells/58046_1737374688393/bin/npm
    pnpm: 9.12.3 - /run/user/1000/fnm_multishells/58046_1737374688393/bin/pnpm
  Browsers:
    Chrome: 131.0.6778.69
```
</details>


**Search keywords**: vite pnpm strict

--- a/packages/x-license/package.json
+++ b/packages/x-license/package.json
@@ -35,7 +35,8 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.25.7",
-    "@mui/utils": "^5.16.6 || ^6.0.0"
+    "@mui/utils": "^5.16.6 || ^6.0.0",
+    "@mui/x-internals": "workspace:*"
   },
   "peerDependencies": {
     "react": "^17.0.0 || ^18.0.0 || ^19.0.0"

